### PR TITLE
Create a configuration variable publish_discovery_info which tweak

### PR DESCRIPTION
**Pull request by verma7**
the DiscoveryInfo.

Resolves https://dcosjira.atlassian.net/projects/CASSANDRA/issues/CASSANDRA-19

### CODE REVIEW

File: cassandra-commons/src/main/java/com/mesosphere/dcos/cassandra/common/tasks/CassandraDaemonTask.java [hunk 1]
```diff
@@ -161,7 +161,8 @@ private CassandraDaemonTask(
                 config.getApplication().getSslStoragePort(),
                 config.getApplication().getRpcPort(),
                 config.getApplication().getNativeTransportPort()),
-            getDiscoveryInfo(capabilities, name, config.getApplication().getNativeTransportPort()),
+            getDiscoveryInfo(config.getPublishDiscoveryInfo(), config.getApplication().getClusterName(),
```
Comment (gabrielhartmann) [inline cassandra-commons/src/main/java/com/mesosphere/dcos/cassandra/common/tasks/CassandraDaemonTask.java]: NIT: Once arguments can't fit on one line, they should all be put on new lines.

File: cassandra-commons/src/main/java/com/mesosphere/dcos/cassandra/common/tasks/CassandraDaemonTask.java [hunk 2]
```diff
@@ -289,7 +290,23 @@ public CassandraDaemonTask updateId() {
...
+            LOGGER.info("Publishing discovery info: ")
+            DiscoveryInfo.Builder discoveryBuilder = DiscoveryInfo.newBuilder()
+                    .setVisibility(DiscoveryInfo.Visibility.EXTERNAL)
+                    .setName(clusterName + "." + nodeName);
+            discoveryBuilder.getPortsBuilder().addPortsBuilder()
+                    .setName("NativeTransport")
+                    .setNumber(nativePort);
+            return discoveryBuilder.build();
+        }
+
```
Comment (gabrielhartmann) [inline cassandra-commons/src/main/java/com/mesosphere/dcos/cassandra/common/tasks/CassandraDaemonTask.java]: No.  The `VIP_NODE_PORT` is the port that clients should use to communicate with the VIP.  It should not be allocated by Mesos.  The nativePort is the port allocated by Mesos.  You can make up whatever port you like for `VIP_NODE_PORT`, minuteman maps it to `nativePort`.

**Comment by gabrielhartmann** [comment]
@mesosphere-ci: retest this please

File: cassandra-commons/src/main/java/com/mesosphere/dcos/cassandra/common/tasks/CassandraDaemonTask.java (hunk 2 above)
Comment (verma7) [inline cassandra-commons/src/main/java/com/mesosphere/dcos/cassandra/common/tasks/CassandraDaemonTask.java]: Got it.

File: cassandra-commons/src/main/java/com/mesosphere/dcos/cassandra/common/tasks/CassandraDaemonTask.java (hunk 1 above)
Comment (verma7) [inline cassandra-commons/src/main/java/com/mesosphere/dcos/cassandra/common/tasks/CassandraDaemonTask.java]: Fixed.

**Pull request merged**